CHANGELOG.md

```diff
@@ -4,6 +4,10 @@
 
 ### Snowpark Python API Updates
 
+#### Improvements
+
+- Improved query generation for `Dataframe.stat.sample_by` to generate a single flat query that scales well with large `fractions` dictionary compared to older method of creating a UNION ALL subquery for each key in `fractions`. To enable this feature, set `session.conf.set("use_simplified_query_generation", True)`.
```

Comment:
just curious on how much perf is improved?

Reply:
I'm writing some microbenchmarks for now but we will get better data when https://github.com/snowflakedb/snowpark-python/pull/3103 gets merged

Reply:
I created a simple microbenchmark [here](https://github.com/snowflakedb/snowpark-python-internal/pull/43/files) and I am reporting end-to-end time of getting the results back in table below:


| size(frac) | E2E time (Unoptimized)| E2E time (Optimized)  |
|----------|------------| ------ |
|10 | 1.91s | 1.36s |
|100 | 142.81s | 5.18s |
|500 | 3367.35s | 13.511s |